client/src/shared/Form/AddAdressText.js

```diff
@@ -1,21 +1,53 @@
 import React, { useEffect, useState } from 'react';
 import Autocomplete from '@material-ui/lab/Autocomplete';
 import TextField from '@material-ui/core/TextField';
-import { connect } from 'formik';
+import { connect as connectFormik } from 'formik';
+import { connect } from 'react-redux';
 import PropTypes from 'prop-types';
+import { addNewPoint } from '../../modules/MapHolder/actions/mapState';
 import {
   getGeocodingOptions,
-  getGeocodingDetails
+  getGeocodingDetails,
+  getReverseGeocoding
 } from '../api';
 
-const AddAdressText = ({ formik, className }) => {
+const AddAdressText = ({
+  formik,
+  className,
+  newPoint,
+  addNewPoint
+}) => {
   const [open, setOpen] = useState(false);
   const [options, setOptions] = useState([]);
   const [value, setValue] = useState(formik.values.address);
   useEffect(() => {
     setValue(formik.values.address);
   }, [formik.values.address]);
 
+  const reverseRequest = async () => {
+    const result = await getReverseGeocoding(newPoint);
+    const {
+      lng,
+      lat
+    } = result.data.results[0].geometry.location;
+    formik.setFieldValue(
+      'address',
+      result.data.results[0].formatted_address
+    );
+    formik.setFieldValue('coordinates', [lng, lat]);
+  };
+
+  useEffect(() => {
+    if (Object.keys(newPoint).length !== 0)
+      reverseRequest();
+  }, [newPoint]);
+
+  useEffect(() => {
+    return () => {
+      addNewPoint({});
+    };
+  }, []);
+
   useEffect(() => {
     let active = true;
     if (value.length > 2) {
```

Comment:
You use active variable = true, and since when component unmount, active = false. But when component mount again active will be true again, so it's variable don't make sense.